### ISSUE TITLE
Stop or pause tab button in right click menu

### ISSUE THREAD

**Issue by jh34ghu43gu** [MrAlex94/Waterfox#2633]
**Is your feature request related to a problem? Please describe.**
Sometimes I'll end up with many tabs open when working on a project and might need to put some "away" for a bit while I work on a sub-project that one of the pages told me I need to do. I do expect to come back to these tabs later during the project, but likely only once so bookmarking seems redundant to me. _The problem I have from this methodology is some websites will continue to use resources or force refresh content_ (can't remember which website was doing this but it happened recently) for a decent amount of time after I stop using them.

**Describe the solution you'd like**
I believe a nice solution to this would be to allow users to force "pause" or stop a tab, taking it out memory and leaving just a blank page with the url (or similar to when restoring a previous session the tab remains unloaded until it is clicked on again). The page could be "unpaused" via the same button or simply refreshed to load the tab once it is needed (or once it is clicked on again). 

**Describe alternatives you've considered**
Bookmarking is the other alternative I have considered but as previously stated I find those to be a bit overkill for a page I'm likely only to ever need one or two more times in the future.

**Additional context**
Mostly thinking of the way tabs function when restoring a session after a restart except being able to force a single tab to go into that state manually. Probably a power user feature and I don't know how easy it would be to implement so I understand if it's low priority/not feasible.

**Comment by adamp01** [comment]
@jh34ghu43gu - This should be trivial to implement, you can test yourself by doing the following (in browser console, `Ctrl+Shift+J`:
```
gBrowser.tabs <-- find index of tab you want to test discarding
gBrowser.discardBrowser(gBrowser.tabs[4]) <--- replace 4 with index of tab you want to discard
```

I can add a new context menu item to enable this functionality, `Unload Tab`. Note: you cannot unload the current selected tab.

Another change that will be incoming is the setting of `browser.tabs.unloadOnLowMemory` to `true`. It may benefit you to change that in `about:config` as well.

**Comment by jh34ghu43gu** [comment]
Thank you!

**Comment by adamp01** [comment]
I implemented this for G4.1.0, let me know if you encounter any issues with it.
